pontoon/base/models.py

```diff
@@ -25,12 +25,101 @@
+        return Sum(
+                Case(
+                    When(translation_query, then=1),
+                    output_field=models.IntegerField(),
+                    default=0
+                    )
+                )
+
+    def get_changed_translations(self, start_date=None, end_date=None):
+        translations = (
+            Translation.objects
+            .exclude(user=None, string=F('entity__string'))
+            .exclude(string=F('entity__string_plural'))
+        )
+        if start_date and end_date:
+            translations = translations.filter(date__range=(start_date, end_date))
+        return translations
+
+    def get_approved(self, start_date=None, end_date=None):
+        return self.get_changed_translations(start_date, end_date).filter(approved=True)
+
+    def get_unapproved(self, start_date=None, end_date=None):
+        return self.get_changed_translations(start_date, end_date).exclude(approved=True, fuzzy=True)
+
+    def get_needs_work(self, start_date=None, end_date=None):
+        return self.get_changed_translations(start_date, end_date).filter(fuzzy=True)
+
+    def get_contributors_with_stats(self, start_date=None, end_date=None, limit=100):
+        """
+        Returns contributors list, sorted by count of their translations.
+        Every user instance has added following properties:
+        * translations_count
+        * translations_approved_count
+        * translations_unapproved_count
+        * translations_needs_work_count
+        Method has been created mainly to improve performance and to optimize
+        count of sql queries during generation of metrics.
+        """
+        def translations_counts(filter=None, exclude=None):
+            """Short helper to avoid duplication of passing dates."""
+            return self._changed_translations_counts(start_date, end_date, filter=filter, exclude=exclude)
+
+        contributors = (
+            self
+            .filter(translation__in=self.get_changed_translations(start_date, end_date)).distinct()
+            .exclude(email__in=settings.EXCLUDE)
+            .annotate(translations_count=self._changed_translations_counts(start_date, end_date),
+                      translations_approved_count=translations_counts(filter=dict(translation__approved=True)),
+                      translations_unapproved_count=translations_counts(exclude=dict(translation__approved=True, translation__fuzzy=True)),
+                      translations_needs_work_count=translations_counts(filter=dict(translation__fuzzy=True)))
+            .exclude(translations_count=0)
+        )
+        return contributors.order_by('-translations_count')[:limit]
+
+
 # User class extensions
 @property
 def user_display_name(self):
     name = self.first_name or self.email.split('@')[0]
     return u'{name} <{email}>'.format(name=name, email=self.email)
+
 User.add_to_class('display_name', user_display_name)
+User.add_to_class('translations', UserTranslationsManager())
```

Comment:
`translations` isn't a good name because it implies that `User.translations` has to deal with querysets containing Translations instead of Users.

Maybe `translators`?